docs/reference/language/context.md

```diff
@@ -25,38 +25,180 @@ in some places that are also aware of their location in the document:
+
+Moreover, some functions, such as [`to-absolute`]($length.to-absolute) 
+and [`counter.display`]($counter.display) are only applicable in a context, 
+because their results depend on the current settings of style properties. 
+When another function `foo()` calls a context-dependent function, it becomes 
+itself context-dependent:
+
+```example
+#let foo() = 1em.to-absolute()
+#context {
+  // foo() cannot be called outside of a context
+  foo() == text.size
+}
+```
+
+When a property is changed, the response to the query changes accordingly: 
 
 ```example
-#let value = context text.lang
-#value
+#set text(lang: "en")
+#context text.lang
 
 #set text(lang: "de")
-#value
+#context text.lang
+```
+
+The output of a `#context ...` call is static in the form of opaque 
+[`content`]. Write access to context output is prohibited, as it would 
+often result in invalid code: If the context changes between read and 
+write, overwriting a property would cause an inconsistent system state. 
+In fact, context-dependent property fields are immutable constants even 
+within the context itself:
+
+```example
+#set text(lang: "en")
+#context [
+  call 1: #text.lang \
+
+  #set text(lang: "fr")
+  call 2: #text.lang
+]
+```
+
+Both calls have the same output 'en', because `text.lang` is assigned
+upon entry in the context and remains constant until the end of its scope 
+(the closing `]`). It does not "see" the `#set text(lang: "fr")` before 
+call 2. Compare this to the previous example: there we got two different 
+results because we created two different contexts.
+
+However, immutability only applies to the property fields themselves. 
+Content creation instructions within a context _do_ see the effect of 
+the set rule. Consider the same example with font size:
+
+```example
+#set text(size: 50pt)
+#context [
+  call 1: #text.size \
+
+  #set text(size: 25pt)
+  call 2: #text.size
+]
+```
+
+Call 2 still outputs '50pt', because `text.size` is a constant.
+However, this output is printed in '25pt' font, as specified by the set
+rule before the call. This illustrates the importance of picking the 
+right insertion point for a context to get access to precisely the right 
+styles. If you need access to updated property fields after a set rule, 
+you can use nested contexts:
+
+```example
+#set text(lang: "en")
+#context [
+  call 1: #text.lang \
+
+  #set text(lang: "fr")
+  call 2: #context text.lang
+]
+```
+
+All of the above applies to `show` rules analogously, for example:
+
+```example
+#let template(body) = {
+  set text(size: 25pt)
+  body
+}
+
+#set text(size: 50pt)
+#context [
+  call 1: #text.size \
+
+  #show: template
+  call 2: #text.size \
+  call 3: #context text.size
+]
+```
+
+## Controlling content creation within a context
+
+The main purpose of retrieving the current values of properties is, 
+of course, to use them in the calculation of derived properties, 
+instead of setting those properties manually. For example, you can
```

Comment:
I wouldn't say that's accurate. It's surely one use case, but you might just as well use it for something else.

Reply:
For example to localize a document via `context if text.lang == "de" [Inhalt] else [Contents]`. I think the use cases are very diverse, which is why I stumbled over the "of course"